packages/@ember/-internals/glimmer/lib/helpers/helper.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ @module ember
+ */
+
+/**
+ Use the `{{helper}}` helper to create contextual helper so
+ that it can be passed around as first-class values in templates.
+
+ ```handlebars
+ {{#let (helper "join-words" "foo" "bar" separator=" ") as |foo-bar|}}
```

Comment:
One more thing (same for the other snippet): `foo-bar` is it actually legal to have a dash in the local variable (block param) name? I probably should know this but I don't 🙃

Reply:
According to AST explorer it seems to be fine so that's good enough for me